File: Algebra_ImproperToMixed.cs
using System;
using System.Collections;

namespace Algebra_CS
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to our demonstration sequels");
            Console.WriteLine("Hope you enjoy (and follow) the lessons.");
            Console.WriteLine("\r\n");

            /*
            * Convert fractions from Mixed to Improper
             */
            Hashtable fraction = new Hashtable();
            fraction["numerator"] = 10;
            fraction["denominator"] = 3;

            Console.WriteLine("    Converting from Improper to Mixed the fraction:");
            // Print as fraction
            Console.WriteLine(String.Format("{0,55}", fraction["numerator"]));
            Console.WriteLine(String.Format("{0,55}", "-"));
            Console.WriteLine(String.Format("{0,55}", fraction["denominator"]));

            // use the MixedToImproper class
            ImproperToMixed imp_mix = new ImproperToMixed(fraction);
            fraction = imp_mix.doConvert();

            Console.WriteLine(Environment.NewLine);

            Console.WriteLine(String.Format("{0,52}", fraction["numerator"]));
            Console.WriteLine(String.Format("{0,48} {1} {2}", "Answer =   ", fraction["whole_number"], "-"));
            Console.WriteLine(String.Format("{0,52}", fraction["denominator"]));

        }
    }
}

File: Algebra_ImproperToMixed.vb
Module Algebra_improper2mixed

    Sub Main()

        ''
        ' Convert fractions from Improper To Mixed
        ''
        Dim fraction As New Dictionary(Of String, Integer)
        fraction.Add("numerator", 10)
        fraction.Add("denominator", 3)

        Console.WriteLine("    Converting from Improper to Mixed the fraction:")
        ' Print as fraction
        Console.WriteLine(String.Format("{0,55}", fraction.Item("numerator")))
        Console.WriteLine(String.Format("{0,55}", "-"))
        Console.WriteLine(String.Format("{0,55}", fraction.Item("denominator")))

        ' use the ImproperToMixed Class
        Dim imp2mix As New ImproperToMixed
        imp2mix._init_(fraction)
        fraction = imp2mix.doConvert()

        Console.WriteLine(Environment.NewLine)

        Console.WriteLine(String.Format("{0,52}", fraction.Item("numerator")))
        Console.WriteLine(String.Format("{0,48} {1} {2}", "Answer =   ", fraction.Item("whole_number"), "-"))
        Console.WriteLine(String.Format("{0,52}", fraction.Item("denominator")))

    End Sub

End Module
